src/Cns.ts

```diff
@@ -57,14 +54,37 @@ export default class Cns extends NamingService {
     );
   }
 
+  static async getNetworkConfigs(config?: { url: string } | { provider: Provider } ): Promise<{ network: CnsSupportedNetworks, provider: Provider } | undefined> {
```

Comment:
I think the logic of this methods should go into Resolution class itself. It's also valid for Ens implementation since there are almost two similar methods.